### DISCUSSION TITLE
How to import a binary stl file to porespy?

### DISCUSSION THREAD

**Discussion by josephinoo** [PMEAL/porespy#700] [Q&A] [answered]
Hello, I want to import a binary stl file, to use it in porespy, is there a possibility?

**Comment by ma-sadeghi** [comment, marked as answer]
Maybe try [this](https://stackoverflow.com/a/31924982).

**Reply by josephinoo** [reply]
How can I pass these points that represent the coordinates of a solid, to a binary matrix. @ma-sadeghi  
Point 0: x
Point  1: y
Point 2 : z
![image](https://user-images.githubusercontent.com/46098690/172034199-ec401426-1349-44e7-944f-5051dc02bfb3.png)

**Reply by ma-sadeghi** [reply]
Hmm, these are I guess the corrdinates of triangular surfaces that make up the solid phase, right?

**Reply by jgostick** [reply]
Are you looking for something [like this](https://github.com/PMEAL/porespy/discussions/680)?

**Reply by josephinoo** [reply]
We solved with this
Thanks @ma-sadeghi @jgostick 

```python
scale = 10**(-7.1)
x = ((d['Points:0'] - d['Points:0'].min() )/scale).astype(int)
y = ((d['Points:1'] - d['Points:1'].min())/scale).astype(int)
z = ((d['Points:2'] - d['Points:2'].min() )/scale).astype(int)
im = np.zeros(shape, dtype=bool)
for i in range(len(x)):
    im[x[i],y[i],z[i]]=True
```
![image](https://user-images.githubusercontent.com/46098690/174951076-3e46403c-7452-48da-8cc9-c50f859ccec6.png)

**Reply by jgostick** [reply]
nice!
